src/harness/vfsUtil.ts

```diff
@@ -650,14 +650,14 @@ export class FileSystem {
      * NOTE: do not rename this method as it is intended to align with the same named export of the "fs" module.
      */
     public readFileSync(path: string, encoding?: BufferEncoding | null): string | Buffer;
-    public readFileSync(path: string, encoding: BufferEncoding | null = null) { // eslint-disable-line no-null/no-null
+    public readFileSync(path: string, encoding: BufferEncoding | null | undefined = undefined) { // eslint-disable-line no-null/no-null
         const { node } = this._walk(this._resolve(path));
         if (!node) throw createIOError("ENOENT");
         if (isDirectory(node)) throw createIOError("EISDIR");
         if (!isFile(node)) throw createIOError("EBADF");
 
-        const buffer = this._getBuffer(node).slice();
-        return encoding ? buffer.toString(encoding) : buffer;
+        const fileBuffer = this._getBuffer(node, encoding ?? undefined);
+        return !fileBuffer.encoding ? fileBuffer.data.slice() : fileBuffer.data;

```

Comment:
```suggestion
        return fileBuffer.encoding ? fileBuffer.data : fileBuffer.data.slice();
```

Reply:
I mistakenly pushed that change to this branch.